app/src/features/surveys/view/survey-animals/form-sections/MeasurementAnimalForm.tsx

```diff
@@ -98,23 +86,31 @@ interface MeasurementFormContentProps {
   measurements?: IMeasurementStub[];
 }
 
-const MeasurementFormContent = ({ index, measurements }: MeasurementFormContentProps) => {
+export const MeasurementFormContent = ({ index, measurements }: MeasurementFormContentProps) => {
   const { values, handleChange, setFieldValue, handleBlur } = useFormikContext<IAnimal>();
-  const [measurement, setMeasurement] = useState<IMeasurementStub>();
-
+  const { animalKeyName } = ANIMAL_SECTIONS_FORM_MAP[SurveyAnimalsI18N.animalMeasurementTitle];
   const taxonMeasurementId = values.measurements[index].taxon_measurement_id;
+  const [measurement, setMeasurement] = useState<IMeasurementStub | undefined>(
+    measurements?.find((m) => m.taxon_measurement_id === taxonMeasurementId)
+  );
   const isQuantMeasurement = has(measurement, 'unit');
 
-  const tMeasurementIDName = getAnimalFieldName<IAnimalMeasurement>(NAME, 'taxon_measurement_id', index);
-  const valueName = getAnimalFieldName<IAnimalMeasurement>(NAME, 'value', index);
-  const optionName = getAnimalFieldName<IAnimalMeasurement>(NAME, 'qualitative_option_id', index);
+  const tMeasurementIDName = getAnimalFieldName<IAnimalMeasurement>(animalKeyName, 'taxon_measurement_id', index);
+  const valueName = getAnimalFieldName<IAnimalMeasurement>(animalKeyName, 'value', index);
+  const optionName = getAnimalFieldName<IAnimalMeasurement>(animalKeyName, 'qualitative_option_id', index);
+
+  useEffect(() => {
+    setMeasurement(measurements?.find((m) => m.taxon_measurement_id === taxonMeasurementId));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [measurements]); //Sometimes will not display the correct fields without this useEffect but could have side effects, may need to revisit.
 
   const handleMeasurementTypeChange = (event: SelectChangeEvent<unknown>) => {
     handleChange(event);
     setFieldValue(valueName, '');
     setFieldValue(optionName, '');
     const m = measurements?.find((m) => m.taxon_measurement_id === event.target.value);
```

Comment:
We should avoid single-letter variable names; I would call this `const measurement`

Reply:
Since `measurement` is already taken, you could rename that to `const [currentMeasurement, setCurrentMeasurement]` or maybe `const [formMeasurement, setFormMeasurement]`

Reply:
fixed